### ISSUE TITLE
xboa migrate throwing fatal error - and daedalus showing a revert to bookworm with barracuda up-lts (jdk-jre ?)

### ISSUE THREAD

**Issue by EdNett** [omega8cc/boa#1837]
Hello,

I am moving to a new vps and the xboa migrate seemed to work, but then I noticed that it had substituted the source octopus o1.fqdn for the target o1.fqdn - which existed (without any platforms and sites) before the migrate. So, this means no sites were imported in to the GUI - but they worked with the new DNS - that is - they resolve. Somehow the xboa migrate all sites also migrated the hostmaster site from the target! - but it isn't actually there it's just a directory and a few files - there is no actual drupal hostmaster site there now, and the old one is gone!

Every time I try to run an octopus up-lts it tries to bring in the source o1.fqdn - there must be a file somewhere controlling this - and it isn't the /root/.o1.octopus.cnf file.

Is there any way to fix this?

boa info more of target is here: https://gist.github.com/EdNett/9e0a1068a09723d2039cbfe00803b3f8

I have changed every instance of o1.source-fqdn on the server (except one, perhaps - that I cannot find) to o1.target-fqdn, and still every time I run an octopus up-lts o1 it echos to the screen that it is doing a o1.source-fqdn update !  Where is that data coming from - a file ? Which ? or is it in the db ? Thanks for pointing me in the right direction. I hope to be able to get this going again.

I can see at the end of the octopus up-lts that it is still trying to rsync with the old hostmaster file on the source server ! ? How to stop this behavior ? I am removing rsync with aptitude and will try to run octopus up-lts again. - this may be because I thought I needed to enable remote import on the target - which I did AFTER the migrate octopus finished. I have now disabled it, and reinstalled rsync - since I see that rsync is necessary even when not importing from a remote source.

note that it shows "bookworm" error after being daedalus for many months - when I ran the last barracuda up-lts, I saw that it was installing jdk jre runtime environment (I think) from bookwork repos - I don't know if that is the problem or not?

Best,

Ed

**Comment by EdNett** [comment]
Even though the remote import module was active for only an hour or so, and it has been disabled for over 12 hours now, when I attempt a hostmaster site or platform verify I get errors showing rsync is still trying to connect to the source server (and no sites or platforms which were migrated appear in the octopus GUI - luckily the sites are still resolving in the browser and are up!):
``
Platforms permissions of /data/disk/o1/platforms have been changed to 755.	
-
success
Platforms path /data/disk/o1/platforms is writable.	
-
success
Could not rsync from '/data/disk/o1/platforms' to 'o1@source-fqdn.com:/'	
-
error
/data/disk/o1/platforms could not be synced to remote server source-fqdn.com. Changes might not be available until this has been done. (error: Host key verification failed.
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(231) [sender=3.2.7])
-
error
Adjusted sync path to match `composer_git_path`.	
-
info
Could not rsync from '/data/disk/o1/config/server_master' to 'o1@source-fqdn.com:/'	
-
error
/data/disk/o1/config/server_master could not be synced to remote server source-fqdn.com. Changes might not be available until this has been done. (error: Host key verification failed.
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(231) [sender=3.2.7])
-
error
Command dispatch complete	
-
notice
Invoking hosting_verify_task_rollback hooks.	
-
notice
Command dispatch complete	
-
notice
Updated task status to "Failed"
``
I don't thing my enabling that module remote import for an hour has anything to do with this problem. How can I stop rsync from trying to connect to the source server when I do a verify task?

**Comment by EdNett** [comment]
I also tried to run a barracuda up-lts to get back to Daedalus, following your upgrade guide, and it failed and I cannot seem to get the server back on Daedalus as far as BOA is concerned.

**Comment by omega8cc** [comment]
Hello,  

Running `barracuda up-lts` will not revert your OS to an older version. The issue you’re experiencing is likely due to an incomplete installation, which was a known random issue affecting Daedalus.  

Please try updating the system with the following command:  

```  
apt-get install base-files=12.4+deb12u7devuan1 --allow-downgrades  
```  

We’ve already updated our autoinit logic to address this issue:  
https://github.com/omega8cc/boa/commit/0dda9815eb5810302f56635b943a240c855114d4  

Let us know if this resolves the issue or if you need further assistance.

**Comment by EdNett** [comment]
Hello,

The apt sources.list are all bookworm now, and so that package can't be found, of course. 
Specifying _BOOKWORM_TO_DAEDALUS=YES  and running up-lts system
starts the process, (it's important to delete any AUTO_TO_DAEDALUS in the barracuda.cnf because that seems to put the upgrade process off track, it seems). Once teh sources.list are devuan not debian then changing the base files as specified above works, of course. It is taking 3 or more up-lts to get the vps back to Daedalus. I wonder:
1) should some of the run-*-*-daedalus.cnf riles in /root/ be deleted ? (since the vps already did the up to daedalus a few months or years ago)? If so, which one?
and
2) should the server provider's repos be removed from /etc/apt/sources.list.d/... since they all specify bookworm ? - or should I let the upgrade process remove them - which it now appears to do (it didn't at first) ?

**Comment by EdNett** [comment]
gist of errors for o1 (which was target of unsuccessfully-completed-migrate platform) is here:
https://gist.github.com/EdNett/6d32dd2fa03e28f5dbe7d9557f4b8a93

Is there any way to get this attepted rsync to stop? 

Or do I have to delete the platform and rebuild sites manually?

**Comment by omega8cc** [comment]
It was a very bad idea to proceed with migration if the target system was visibly broken.

We would recommend to re-activate Aegir on the old server and run sites verify to disable any proxy and then recreate the new server from scratch before attempting another migration.

**Comment by EdNett** [comment]
Hello. I followed your advice and tried the migration onto an unused vps which did not have an o1 octopus installed. I noticed that during the process there were two errors:

1) INFO: export for o1 to target-ip  complete
server1:~# xboa transfer o1 target-ip
_USR o1 is not eligible for migration: already transferred


and

2) during the "import", that is when on the target host and running "xboa import o1 target-ip"
for each and every site being impoted there was an error like this:
INFO: site domain.com db import on o1 complete
Failed to load or create a task for ID '@domain.com'.               [error]
INFO: site domain.com verify on o1 scheduled


and of course, there are no static platforms and no sites except the octopus hostmaster on the newly created o1 GUI.

Should I expect those platforms and sites, which DO exist in the target host's file structure in the right place, to magically appear in the octopus GUI ? Or do I need to manually create tasks so that they appear? (I don't believe I should manually create the platforms and sites within them on teh Octopus GUI - that would be wrong, wouldn't it?

Thanks for any help. Both the Source and the Target are running the exact same version of everything (possibly a php mismatch since I don't know what php version the newly created target o1 octopus was created with) adn both were repeatedly put up-to-date before doing the migration.

Is this a failed migration because the sites aren't showing up along with the platforms in the Octopus GUI?

Lastly one setp I didn't do was on the souurce host at teh end:  "xboa proxy o1 target-ip"  - I don't want to move the dns until I'm sure that the target o1 octopus platform will work properly. I can modify IPv4 at the nameservers quickly as needed. This shouldn't cause the target o1 newly-created octopus instance to fail to import the static platforms and sites, right?
